### PR TITLE
retire our own cup-packaging

### PR DESCRIPTION
**Pull request by lsf37**
* use `com.github.vbmacher:java-cup{-runtime}` instead
 * update mvn and bazel deps
 * move cup-sample-project into cup-maven-plugin

This should enable a JFlex build from source from a release package outside our own repo.

### CODE REVIEW

**Comment by lsf37** [comment]
Ok, this is now ready for review.

Having `java-cup-runtime` and `java-cup` as external dependency in maven and bazel seems to be working fine and fixes the release package problem. It looks like I already did use that dependency for `ant` in the examples and had forgotten about it, so they're now actually all using the same setup.

**Comment by regisd** [comment]
I'm fine with it. Just want to point out that 
- vbmacher is not the official cup maintainer, and may stop doing this in the future (the code is  hosted on from https://github.com/vbmacher/cup-maven-plugin)
- I thought you wanted to keep track of the cup code to see diffs across versions
- For Bazel, an alternative could be to send a PR with the BUILD file to https://github.com/vbmacher/cup-maven-plugin

**Comment by lsf37** [comment]
> * vbmacher is not the official cup maintainer, and may stop doing this in the future (the code is  hosted on from https://github.com/vbmacher/cup-maven-plugin)

Yes, he even says it's not actively maintained apart from tracking CUP releases, although he is taking pull requests. We may need to change again for the next CUP release, but it looks like they are rare now.

> * I thought you wanted to keep track of the cup code to see diffs across versions

Indeed, that is the disadvantage of this solution. It was more important to me when there were still CUP releases every year or so, but I might still regret dropping the source from our repo when the next one comes out. At least CUP is now using git and quoting the hash for their release, so it's a bit easier to keep track of what's changing. I'm willing to risk it. The alternative would be to add source for our own `java_cup` package and publish that, but it looks like that will be more work than is justified by the benefit of the diff.

> * For Bazel, an alternative could be to send a PR with the BUILD file to https://github.com/vbmacher/cup-maven-plugin

Instead of adding it to `third_party`? We could still do that later after 1.8.1.

**Comment by lsf37** [comment]
> I guess a follow-up step is to remove our version of cup-maven-plugin and use vbmacher's one instead

I tried using it, but it's a bit awkward, i.e. needs explicit naming of `className` and `cupDefinition`, and doesn't seem to be able to deal with classes without package as we use in the examples.

We could raise PRs for that based on our own plugin, though, and make it work. The vmbacher plugin does support pretty much all options that CUP provides, so that might indeed be the shortest path to a cup plugin with support for everything we want.

File: third_party/com/github/vbmacher/java-cup-runtime/BUILD.bazel [hunk 1]
```diff
@@ -0,0 +1,13 @@
+# Java Cup Runtime
+# repackaging of runtime classes of JavaCup parser generator
+# https://github.com/vbmacher/java-cup
```
[the remark this answers is not in the document]
Comment (lsf37) [reply third_party/com/github/vbmacher/java-cup-runtime/BUILD.bazel]: Indeed, thanks for checking. Now fixed.

File: third_party/com/github/vbmacher/java-cup/BUILD.bazel [hunk 2]
```diff
@@ -0,0 +1,13 @@
+# Java Cup
+# repackaging of JavaCup parser generator
+# https://github.com/vbmacher/java-cup
```
[the remark this answers is not in the document]
Comment (lsf37) [reply third_party/com/github/vbmacher/java-cup/BUILD.bazel]: Fixed, thanks.

**Pull request merged**